src/_guides/language/coming-from/swift-to-dart.md

```diff
@@ -0,0 +1,2982 @@
+const double atm = 1.01325 * bar; // Standard atmosphere
+```
+
+A `const` variable defined at a class level
+needs to be marked as `static const`.
+
+{:.include-lang}
+```dart
+class StandardAtmosphere {
+  static const bar = 1000000; // Unit of pressure (dynes/cm2)
+  static const double atm = 1.01325 * bar; // Standard atmosphere
+}
+```
+
+The `const` keyword isn't just for declaring
+constant variables; it can also be used to create
+constant values:
+
+{:.include-lang}
+```dart
+var foo = const ['one', 'two', 'three'];
+foo.add('four'); // Error: foo contains a constant value.
+foo = ['apple', 'pear']; // This is allowed as foo itself isn't constant.
+foo.add('orange'); // Allowed as foo no longer contains a constant value.
+```
+
+In the above example, you can't change the `const` value
+(add, update, or remove the elements in the given list),
+but you can assign a new value to `foo`.
+After `foo` is assigned a new (non-constant) list,
+you _can_ add, update, or remove the contents of the list. 
+
+You can also assign a constant value to a `final` field.
+You can't use the `final` field in a constant context,
+but you can use the constant. For example:
+
+{:.include-lang}
+```dart
+final foo1 = const [1, 2, 3];
+const foo2 = [1, 2, 3]; // Equivalent to `const [1, 2, 3]`
+const bar2 = foo2; // OK
+const bar1 = foo1; // Compile-time error, `foo1` isn't constant
+```
+
+You can also define `const` constructors,
+making those classes immutable (unchanging)
+and making it possible to create instances of
+those classes as compile-time constants.
+For more information, check out
+[const constructors](#const-constructors).
+
+## Built-in types
+Dart includes a number of types in the platform
+libraries, such as:
+
+* Basic value types like
+  * Numbers (`num`, `int`, `double`)
+  * Strings (`String`)
+  * Booleans (`bool`)
+  * The value null (`Null`)
+* Collections
+  * Lists/arrays (`List`)
+  * Sets (`Set`)
+  * Maps/dictionaries (`Map`)
+
+For more information, check out [Built-in types][]
+in the Dart language tour.
+
+[Built-in types]: /guides/language/language-tour#built-in-types
+
+### Numbers
+
+Dart defines three numeric types for holding numbers:
+
+`num`
+: A generic 64-bit number type.
+
+`int`
+: A platform-dependent integer number.
+  In native code, it's a 64-bit two's complement integer.
+  On the web, it's a non-fractional 64-bit,
+  floating-point number.
+
+`double`
+: A 64-bit floating point number.
+
+Unlike Swift, there are no specific types for
+unsigned integers.
+
+All these types are also classes in the Dart API.
+Both the `int` and `double` types share `num`
+as their parent class:
+
+<img
+  src="/assets/img/number-class-hierarchy.svg"
+  alt="Object is the parent of num, which is the parent of int and double">
+
+As number values are technically class instances,
+they have the convenience of exposing their own
+utility functions. Because of this, a `double` can,
+for example, be rounded up as follows:
```

Comment:
The following code snippets are not talking about rounding up a double, but rather converting an integer to a double. Would be nice to include the rounding example though :)

Reply:
I'm going to leave this to you. I finished this article as a favor to Mit. But I've told Tony that once, I land this PR, it's all his.  :D